### DISCUSSION TITLE
Merge LAS files, partially

### DISCUSSION THREAD

**Discussion by issimo007** [kinverarity1/lasio#528] [Q&A] [answered]
Hello, I love LASIO so thank you!  Quick question: although I can easily perform this task by transforming the data into a Pandas DataFrame, is there a way to merge 2 LAS files?  Specifically, say file1 has 10 curves and file 2 has 8 curves similar to file 1 plus 3 new curves?  In a df this is no problem but I don't know how to go back and make an LAS file out of that DataFrame.  Thank you!

**Comment by ThomasMGeo** [comment, marked as answer]
After merging the dataframes in pandas:

https://stackoverflow.com/questions/53645882/pandas-merging-101

You can create a .las using this function:
LASFile.set_data_from_df(df, **kwargs)

 Documentation [here](https://lasio.readthedocs.io/en/latest/lasio.html?highlight=set_data#lasio.LASFile.set_data_from_df)

**Reply by ThomasMGeo** [reply]
If it's only a handful of curves, and you have them in a dataframe, you can also add a curve:

LASFile.insert_curve(ix, mnemonic, data, unit='', descr='', value='')

**Reply by issimo007** [reply]
Thank you!
